chrome/content/zotero/itemPane.js

```diff
@@ -185,6 +189,132 @@ var ZoteroItemPane = new function() {
+			_translationTarget = Zotero.Libraries.userLibrary;
+		}
+		for (let item of items) {
+			yield item.translate(_translationTarget.libraryID, collectionID);
+		}
+	});
+	
+	
+	this.buildTranslateSelectContextMenu = function (event) {
+		_blockTranslate = true;
+		var menu = document.getElementById('zotero-item-addTo-menu');
+		if (event.target != menu) {
+			return;
+		}
+		// Clear previous items
+		while (menu.firstChild) {
+			menu.removeChild(menu.firstChild);
+		}
+		
+		var libraries = Zotero.Libraries.getAll();
+		for (let library of libraries) {
+			if (!library.editable || library.libraryType == 'publications') {
+				continue;
+			}
+			let libraryMenu = _createTranslationMenuForTarget(library);
+			menu.appendChild(libraryMenu);
+		}
+	};
+	
+	
+	this.setTranslateButton = function(name) {
+		var label = Zotero.getString('general.addTo', name ? name : _translationTarget.name);
+		var elem = document.getElementById('zotero-feed-item-addTo-button');
+		elem.setAttribute('label', label);
+
+		var objectType = name ? 'library' : _translationTarget._objectType;
+		var imageSrc = _getCollectionImageSrc(objectType);
+		elem.setAttribute('image', imageSrc);
+	};
+	
+
+	this.setTranslationTarget = function(translationTarget) {
+		_translationTarget = translationTarget;
+		ZoteroItemPane.setTranslateButton()
+	};
+	
+		
+	this.setToggleReadLabel = function() {
+		var markRead = false;
+		var items = ZoteroPane_Local.itemsView.getSelectedItems();
+		for (let item of items) {
+			if (!item.isRead) {
+				markRead = true;
+				break;
+			}
+		}
+		var elem = document.getElementById('zotero-feed-item-toggleRead-button');
+		if (markRead) {
+			elem.setAttribute('label', Zotero.getString('pane.item.markAsRead'));
+		} else {
+			elem.setAttribute('label', Zotero.getString('pane.item.markAsUnread'));
+		}
+	};
+
+
+	/**
+	 * Create a libraryOrCollection DOM tree to place in <menupopup> element.
+	 * 
+	 * @return {Node<menuitem>/Node<menu>}
+	 */
+	function _createTranslationMenuForTarget(libraryOrCollection) {
+		var ns = "http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul";
+		function _createMenuitem(label, value, icon, command) {
+			let menuitem = document.createElementNS(ns, 'menuitem');
+			menuitem.setAttribute("label", label);
+			menuitem.setAttribute("value", value);
+			menuitem.setAttribute("image", icon);
+			menuitem.addEventListener('command', command);
+			return menuitem
+		}	
+		
+		function _createMenu(label) {
+			let menu = document.createElementNS(ns, 'menu');
+			menu.setAttribute("label", label);
+			menu.label = label;
+			menu.type;
+			let menupopup = document.createElementNS(ns, 'menupopup');
+			menu.appendChild(menupopup);
+			return menu;
+		}
+		
+		var imageSrc = _getCollectionImageSrc(libraryOrCollection._objectType);
+		var menuitem = _createMenuitem(libraryOrCollection.name, libraryOrCollection.id, imageSrc,
+				ZoteroItemPane.setTranslationTarget.bind(this, libraryOrCollection));
```

Comment:
Instead of the `_blockTranslate` stuff above, you can just do this:

```
		var menuitem = _createMenuitem(
			libraryOrCollection.name,
			libraryOrCollection.id,
			imageSrc,
			function (event) {
				ZoteroItemPane.setTranslationTarget(libraryOrCollection);
				event.stopPropagation();
			}
		);
```

`setTranslationTarget` will trigger but not the command on the button.